fix: use the meetings preferences api for webex site

Neeraj-swarnkar opened the pull request:
<!--
Hey there,\
Thank you for taking the time to improve our code! 🙂\
Please let us know why this change is necessary and what testing you have performed. \
This ensures our reviewers understand the impact of your change. \

**IMPORTANT**
FAILING TO FILL OUT THIS TEMPLATE WILL RESULT IN REJECTION OF YOUR PULL REQUEST
This is for compliance purposes with FedRAMP program.
-->

# COMPLETES # [SPARK-325626](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-325626) - Start Using the /meetings api for preferred webex site 

## This pull request addresses

JIRA Ticket - [SPARK-325626](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-325626) - Start Using the /meetings api for preferred webex site 

Started using the meetings API for the preferred Webex site
Changed the function name to getMeetingPreferences
Used the https://webexapis.com/v1/meetingPreferences and mention it in the constant file
Updated the util function to parse the site info and get the default Webex site from it
Unit Testing covered

## by making the following changes

I started using the meetings API for the preferred Webex site
Moved From using identity to fetch Preferred Webex site to https://developer.webex.com/docs/api/v1/meeting-preferences/get-meeting-preference-details
For fetch preferred Webex site information

Use the above api to fetch the preferred meeting site rather then using the identity

<!-- You may include screenshots -->

### Change Type

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

arun3528 reviewed (commented):
looks good we will have to test adhoc meeting

Neeraj-swarnkar commented:
@arun3528, I have fixed the Unit test failure issues, 
We have a Green Build. 
Kindly review. Thanks.

arun3528 reviewed (approved):
LGTM but did we test the adhoc meeting scenario 

Create a new space and then start a adhoc meeting by doing 

meetings.create("space conversation url' ,'CONVERSATION_URL')

You can get the spaceconversation URl from web.webex.com

Neeraj-swarnkar commented:
> looks good thanks for looking into it

Hi @arun3528 , We have a green build can you please merge.

arun3528 commented:
@Neeraj-swarnkar  can we squash the commits

Neeraj-swarnkar commented:
> @Neeraj-swarnkar can we squash the commits Also add the plugin-meeting prefix

@arun3528, Updated the prefix in the [commit ](https://github.com/webex/webex-js-sdk/pull/2357/commits/048205a673a2a79c180165c575482c32c6100c76),

Squashed the commits, Build is running and will update once green build without issue

Neeraj-swarnkar commented:
> 

@arun3528, Incorporated the comments, Please re-review and merge. Thanks.

arun3528 commented:
@marcin-bazyl  can you have a quick look at it before we merge

marcin-bazyl commented on packages/node_modules/@webex/plugin-meetings/src/meetings/util.js:
```diff
@@ -65,23 +65,15 @@ MeetingsUtil.checkForCorrelationId = (deviceUrl, locus) => {
 };
 
 MeetingsUtil.parseUserPreferences = (userPreferences) => {
```
the method name is "parseUserPreferences" but it returns jus a single thing - the siteUrl, so either we should either:
1. rename it to say "parseSiteUrlFromUserPreferences" or
2. change it to return an object that contains the parsed information and for now it might have just 1 field, that's OK, it will probably grow in the future when we start using other fields.

personally I would prefer the 2nd option

Neeraj-swarnkar replied:
Thanks for the comment @marcin-bazyl, Let me check it and update you.

Neeraj-swarnkar replied:
Renaming it as `parseSiteUrlFromUserPreferences` this util is specific to the Webex Sites.

marcin-bazyl commented on packages/node_modules/@webex/plugin-meetings/src/meetings/util.js:
```diff
@@ -65,23 +65,15 @@ MeetingsUtil.checkForCorrelationId = (deviceUrl, locus) => {
 MeetingsUtil.parseUserPreferences = (userPreferences) => {
-  let webexSite = null;
+  let result = null;
+  // filter the user preferences to only have default true
```
also the "default" field is only really applicable to the sites, not for all preferences, right? so the filtering should be done only on sites

Neeraj-swarnkar replied:
Let me update the comment..

Neeraj-swarnkar replied:
@marcin-bazyl  Incorporated the changes.

Neeraj-swarnkar commented on packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/utils.js:
```diff
@@ -1,16 +1,23 @@
-          'SparkTOSAccept',
-          '"preferredWebExSite":"go.webex.com"'
-        ]);
+      it('should return the first default siteUrl from sites', () => {
```
API  says sites will have all available Webex sites and the default site of a user, based on earlier API preferred sites could be more than one. @arun3528 can comment on this if my understanding is correct ? That's why I am checking the first default siteURL.

Neeraj-swarnkar replied:
@arun3528, can there ever be more than one default site? please confim.

Neeraj-swarnkar replied:
@marcin-bazyl  Incorporated the changes. Updated the title of the test.
